core/src/filed/backup.cc

```diff
@@ -1604,7 +1604,12 @@ bool EncodeAndSendAttributes(JobControlRecord* jcr,
 
   {
     std::unique_lock l(jcr->mutex_guard());
-    jcr->JobFiles++;                   /* increment number of files sent */
+    int IncreasedJobFiles;
+    if (__builtin_sadd_overflow(jcr->JobFiles, 1, &IncreasedJobFiles)) {
+      Jmsg0(jcr, M_FATAL, 0, T_("jcr->JobFiles overflow prevented!\n"));
+      return false;
+    }
+    jcr->JobFiles = IncreasedJobFiles; /* increment number of files sent */
```

Comment:
```suggestion
    if (jcr->JobFiles == std::numeric_limits<decltype(jcr->JobFiles)>::max()) {
      Jmsg0(jcr, M_FATAL, 0, T_("You are trying to back up too many files at once.  Please split up your fileset.\n"));
      return false;
    }
    jcr->JobFiles += 1; /* increment number of files sent */
```

Reply:
`__builtin_sadd_overflow` is a gcc builtin and as such not available on windows.   `jcr->JobFiles` is also unsigned, why are we using a simple int here ? If we want to ensure that there is no overflow even with a signed value we should use something like
```
static_cast<std::make_signed_t<decltype(jcr->JobFiles)>>(jcr->JobFiles) >= 0
```
after the increment.

Reply:
I'd even go so far to use a macro like this
```
#if not defined JOB_FILES_MAX
#define JOB_FILES_MAX std::numeric_limits<decltype(JobControlRecord::JobFiles)::max())
#endif
```
Because then we could build with `-DJOB_FILES_MAX=20` and run a proper test.